modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/ButtonTag.java

```diff
@@ -213,42 +225,63 @@ protected int processStartTag() throws Exception {
 		if (Validator.isNotNull(_icon) || Validator.isNotNull(_label)) {
 			JspWriter jspWriter = pageContext.getOut();
 
-			if (Validator.isNotNull(_icon)) {
+			if (Validator.isNotNull(_label)) {
```

Comment:
This way you can't have buttons only with icon (no labels) and I think this should be still possible. 
I would prefer: 
```
if (Validator.isNotNull(_icon) && !iconRight) {...}
if (Validator.isNotNull(_label)) {...}
if (Validator.isNotNull(_icon) && iconRight) {...}
```

And maybe we can extract the code that renders the icon to a private method. Let's see if anyone from backend thinks the same

Reply:
Sounds good.